src/cmake/thirdparty/SetupGeosxThirdParty.cmake

```diff
@@ -194,392 +242,260 @@ if(ENABLE_CALIPER)
+# PETSC
+###############################
+if(DEFINED PETSC_DIR)
+    message(STATUS "PETSC_DIR = ${PETSC_DIR}")
+
+    find_and_register(NAME petsc
+                      INCLUDE_DIRECTORIES ${PETSC_DIR}/include
+                      LIBRARY_DIRECTORIES ${PETSC_DIR}/lib
+                      HEADER petscvec.h
+                      LIBRARIES petsc
+                      DEPENDS metis superlu_dist blas lapack)
+
+    set(ENABLE_PETSC ON CACHE BOOL "" FORCE)
+    set(thirdPartyLibs ${thirdPartyLibs} petsc)
+else()
+    if(ENABLE_PETSC)
+        message(WARNING "ENABLE_PETSC is ON but PETSC_DIR isn't defined.")
     endif()
 
-    blt_register_library( NAME superlu_dist
-                          DEPENDS_ON parmetis metis lapack blas
-                          INCLUDES ${SUPERLU_DIST_INCLUDE_DIRS} 
-                          LIBRARIES ${SUPERLU_DIST_LIBRARY}
-                          TREAT_INCLUDES_AS_SYSTEM ON )
-
-    set( thirdPartyLibs ${thirdPartyLibs} superlu_dist )
+    set(ENABLE_PETSC OFF CACHE BOOL "" FORCE)
+    message(STATUS "Not using PETSc")
 endif()
 
 ################################
-# SUITESPARSE
+# VTK
 ################################
-if( ENABLE_HYPRE )
-    message( STATUS "setting up HYPRE" )
-
-    set(HYPRE_DIR ${GEOSX_TPL_DIR}/hypre)
-
-    find_path( HYPRE_INCLUDE_DIRS HYPRE.h
-            PATHS  ${HYPRE_DIR}/include
-            NO_DEFAULT_PATH
-            NO_CMAKE_ENVIRONMENT_PATH
-            NO_CMAKE_PATH
-            NO_SYSTEM_ENVIRONMENT_PATH
-            NO_CMAKE_SYSTEM_PATH)
-
-    find_library( HYPRE_LIBRARY NAMES HYPRE
-                PATHS ${HYPRE_DIR}/lib
-                NO_DEFAULT_PATH
-                NO_CMAKE_ENVIRONMENT_PATH
-                NO_CMAKE_PATH
-                NO_SYSTEM_ENVIRONMENT_PATH
-                NO_CMAKE_SYSTEM_PATH)
-
-    message(STATUS "HYPRE_INCLUDE_DIRS = ${HYPRE_INCLUDE_DIRS}" )
-    message(STATUS "HYPRE_LIBRARY = ${HYPRE_LIBRARY}" )
-    include(FindPackageHandleStandardArgs)
-    find_package_handle_standard_args(HYPRE  DEFAULT_MSG
-                                    HYPRE_INCLUDE_DIRS
-                                    HYPRE_LIBRARY )
-    if (NOT HYPRE_FOUND)
-        message(FATAL_ERROR "HYPRE not found in ${HYPRE_DIR}. Maybe you need to build it")
-    endif()
-    
-    set( HYPRE_DEPENDS "blas;lapack" )
-    if( ENABLE_SUPERLU_DIST )
-        list( APPEND HYPRE_DEPENDS "superlu_dist" )
-    endif()
+if(DEFINED VTK_DIR)
+    message(STATUS "VTK_DIR = ${VTK_DIR}")
+    find_package(VTK REQUIRED
+                 PATHS ${VTK_DIR})
 
-    blt_register_library( NAME hypre
-                          DEPENDS_ON ${HYPRE_DEPENDS}
-                          INCLUDES ${HYPRE_INCLUDE_DIRS}
-                          LIBRARIES ${HYPRE_LIBRARY}
-                          TREAT_INCLUDES_AS_SYSTEM ON )
+    set(thirdPartyLibs ${thirdPartyLibs} vtk)
+else()
+    if(ENABLE_VTK)
+        message(WARNING "ENABLE_VTK is ON but VTK_DIR isn't defined.")
+    endif()
```

Comment:
Actually you add a lib to the `thirdPartyLibs` list, based on whether the directory exists or not (I think), and without checking for `ENABLE_XXX`?
I would check for `ENABLE_XXX` first and then for the directory where I'm supposed to find it?

If I explicitly ask for `-DENABLE_XXX:BOOL=FALSE`, I think that `XXX` gets added nevertheless. Am I correct?

Reply:
If you don't want to use `XXX` don't define `XXX_DIR`. Also see `host-configs/tpls.cmake` which is now included by all standard host configs.

Reply:
Correct. The idea is that we build the TPLs that we want to use. AFAIK we use VTK on every platform so I didn't add in the logic to disable it.